.github/workflows/e2e-detox.yml

```diff
@@ -0,0 +1,214 @@
+name: Detox E2E Tests
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Coming back on this, one additional consideration is that approach risks hiking our CI costs, since each merge to main will trigger a full E2E run.
On the monorepo, instead of running the E2E test on each merge to the `main` branch, we instead have a daily E2E run against the master branch.

I would propose removing the `main` branch from the `on.push` trigger, and create a `schedule` trigger instead, for testing the main branch regularly.